### PR TITLE
pass at ember2.0 compatibility

### PR DESCRIPTION
**Pull request by gabesmed**
- [x] remove beforeObserver in CollectionLayer
- [x] remove beforeObserver in ArrayPathLayer
- [x] remove beforeObserver in PopupLayer
- [x] remove all ArrayControllers
- [x] ~~remove use of Ember.View from PopupMixin~~ Removed popupViewClass functionality for now. @miguelcobain will replace with popupComponent PR!
- more?...

definitely could use help testing this in as many folks as possible's actual use cases before merging!

### CODE REVIEW

**Comment by twokul** [comment]
:+1:

**Comment by gabesmed** [comment]
how would people feel if we just deprecated popupViewClass functionality for now in order to get Ember2.0 compatibility, and add it back later?

File: addon/layers/array-path.js [hunk 1]
```diff
@@ -70,14 +66,12 @@ export default PathLayer.extend({
       locationsProperty = get(this, 'locationsProperty');
     if(!content) { return; }
 
+    var observers = [];
```
Comment (twokul) [inline addon/layers/array-path.js]: should probably use `const`?

**Comment by miguelcobain** [comment]
I actually have some code that overrides popups to use components.
Can post it tomorrow.

**Comment by gabesmed** [comment]
nice

File: addon/layers/collection.js [hunk 2]
```diff
@@ -25,24 +25,27 @@ export default ContainerLayer.extend({
...
-    const len = content ? get(content, 'length') : 0;
-    this.arrayWillChange(content, 0, len);
-  }),
-
   _contentDidChange: Ember.observer('content', function() {
-    const content = get(this, 'content');
-    if(content) { content.addArrayObserver(this); }
-    const len = content ? get(content, 'length') : 0;
+    this._clear();
+    var content = get(this, 'content');
```
Comment (twokul) [inline addon/layers/collection.js]: same here, `const`?

File: addon/layers/collection.js [hunk 3]
```diff
@@ -25,24 +25,27 @@ export default ContainerLayer.extend({
...
+    if (!content) { return; }
+    content.addArrayObserver(this);
+    this._content = content;
+    var len = get(content, 'length');
```
Comment (twokul) [inline addon/layers/collection.js]: same here, `const`?

**Comment by twokul** [comment]
@gabesmed not sure tbh, we just started to use this addon again so can't report any issues for now

**Comment by gabesmed** [comment]
* * * WE HAVE GREEN * * *

anyone wanna give this a review?

**Comment by twokul** [comment]
@gabesmed LGTM

**Pull request merged**